qemu/patches/Fix-TOLUD-for-PCI-passthroughed-devices.patch

```diff
@@ -0,0 +1,92 @@
+From b2499f140ef9773beadf968b4829e821b7dd1922 Mon Sep 17 00:00:00 2001
+From: =?UTF-8?q?Mateusz=20Pi=C3=B3rkowski?= <mati7337@protonmail.ch>
+Date: Sat, 26 Mar 2022 03:06:35 +0100
+Subject: [PATCH] Fix TOLUD for PCI passthroughed devices
+
+---
+ hw/i386/xen/xen-hvm.c | 60 ++++++++++++++++++++++++++++++++++++++++++-
+ 1 file changed, 59 insertions(+), 1 deletion(-)
+
+diff --git a/hw/i386/xen/xen-hvm.c b/hw/i386/xen/xen-hvm.c
+index 9b432773f0..74648ef11b 100644
+--- a/hw/i386/xen/xen-hvm.c
++++ b/hw/i386/xen/xen-hvm.c
+@@ -198,6 +198,52 @@ qemu_irq *xen_interrupt_controller_init(void)
+ 
+ /* Memory Ops */
+ 
++struct iomem_entry{
++    uint64_t addr_start;
++    uint64_t addr_end;
++    uint64_t name_len;
++    char* name;
++};
++
++void parse_iomem_entry(char* line, struct iomem_entry* entry)
++{
++    uint64_t pos = 0;
++
++    sscanf(line, "%llx-%llx : %lln", &entry->addr_start, &entry->addr_end, &pos);
++    entry->name = line + pos;
++    sscanf(line+pos, "%*s%lln\n", &entry->name_len);
++}
++
++uint64_t get_min_pci_addr()
++{
++    FILE* iomem = fopen("/proc/iomem", "r");
++
++    uint64_t iomem_line_bufsize = 128;
++    char *iomem_line = (char *)malloc(iomem_line_bufsize * sizeof(char));
++
++    uint64_t min_addr = 0;
++
++    while(getline(&iomem_line, &iomem_line_bufsize,iomem) != -1){
++        struct iomem_entry entry;
++        parse_iomem_entry(iomem_line, &entry);
++        if (entry.name == 0) {
```

Comment:
You need to initialize `entry` above to rely on this. But better make `parse_iomem_entry` return success/failure explicitly and use that.